templates/tonic-pyml-deployment.yaml

```diff
@@ -1,128 +1,139 @@
-            mountPath: /var/log/nginx
-          - name: var-log-superd
-            mountPath: /var/log/supervisor
-          - name: var-run-nginx
-            mountPath: /var/run/nginx
-          - name: var-run-superd
-            mountPath: /var/run/supervisor
-          - name: var-cache-nginx
-            mountPath: /var/cache/nginx
-          - name: vector
-            mountPath: /tonic/vector_data
-          - name: public-log
-            mountPath: /publish/logs_public
-          - name: log
-            mountPath: /publish/logs
-          {{- end }}
-        ports:
-        - containerPort: 7700
-        resources:
-          {{- with .Values.resources.pyml_service }}
-          {{- toYaml . | nindent 10 }}
-          {{- end }}
-        startupProbe:
-          httpGet:
-            path: /health
-            port: 7700
-            scheme: HTTPS
-          initialDelaySeconds: 5
-          periodSeconds: 60
-          timeoutSeconds: 30
-          failureThreshold: 5
-      restartPolicy: Always
-      imagePullSecrets:
-      - name: tonicai-build-writer-pull-secret
-      serviceAccountName: {{ template "tonic.serviceAccountName" . }}
-      volumes:
-        {{- if .Values.readOnlyRootFilesystem }}
-        - name: public-log
-          emptyDir: {}
-        - name: var-run-nginx
-          emptyDir: {}
-        - name: var-run-superd
-          emptyDir: {}
-        - name: var-log-nginx
-          emptyDir: {}
-        - name: var-log-superd
-          emptyDir: {}
-        - name: vector
-          emptyDir: {}
-        - name: var-cache-nginx
-          emptyDir: {}
-        - name: log
-          emptyDir: {}
-        - name: tmp
-          emptyDir: {}
-        {{- end }}
-      tolerations:
-      {{- if (and (((.Values.resources).pyml_gpu).tolerations) (((.Values.resources).pyml_gpu).amount)) -}}
-      {{ toYaml .Values.resources.pyml_gpu.tolerations | nindent 8 }}
-      {{- end }}
+{{- /* support values file w/o resource definition */}}
+{{- $resources := coalesce
+  ((((.Values).tonicai).pyml_service).resources)
+  (dict "limits" (dict "memory" "8Gi") "requests" (dict "memory" "512Mi" "ephemeral-storage" "1Gi")) -}}
+{{- $legacyGpuAmount := (((.Values).resources).pyml_gpu).amount -}}
+{{- $newGpuAmount := (get $resources.limits "nvida.com/gpu") }}
+{{- $gpuAmount := coalesce $newGpuAmount $legacyGpuAmount 0 -}}
+{{- if ($gpuAmount) -}}
+{{ $_ := set $resources.limits "nvida.com/gpu" $gpuAmount -}}
+{{- end -}}
+{{- $tolerations := (coalesce
+  ((((.Values).resources).pyml_gpu).tolerations)
+  ((((.Values).tonicai).pyml_service).tolerations)) -}}
+{{ $image := "" }}
+{{- if ((.Values.tonicai).pyml_service).image }}
+{{- $image = .Values.tonicai.pyml_service.image }}
+{{- else }}
+  {{- if .Values.useUnprivilegedContainers }}
+  {{- $image = "quay.io/tonicai/tonic_pyml_service_unprivileged" }}
+  {{- else }}
+  {{- $image = "quay.io/tonicai/tonic_pyml_service" }}
+  {{- end }}
+{{- end }}
```

Comment:
GitHub thinks the whole file changed when this (and the usage below) is really what is up. Similar to above, resources are parameterized under the service key, but this service is slightly more complicated because of the optional use of a GPU resource.